tests/cibyl/unit/sources/elasticsearch/test_api.py

```diff
@@ -348,6 +348,31 @@ def test_get_tests_filter_by_tests_name(self: object,
             1
         )
 
+    @patch.object(ElasticSearch, '_ElasticSearch__query_get_hits')
+    def test_get_tests_jobs_filtered_no_tests(self: object,
+                                              mock_query_hits: object) -> None:
+        """Tests internal logic :meth:`ElasticSearch.get_tests`
+            is correct and filters by test name supporting regex, if some jobs
+            turn out to have no tests as a result of the filtering, they should
+            not be returned.
+        """
+        mock_query_hits.return_value = self.tests_hits
+
+        builds_kwargs = MagicMock()
+        builds_value = PropertyMock(return_value=[])
+        type(builds_kwargs).value = builds_value
```

Comment:
If you what you want is for the mock to have a certain type, I just learnt this morning that you can do that with: Mock(spec=MyClass). That way it will pass 'isinstance' checks.

Reply:
Missed this comment yesterday. Thank you, I did not know that and looks like it can simplify the tests here.